webrender_api/src/display_item.rs

```diff
@@ -89,6 +89,58 @@ impl LayerPrimitiveInfo {
             tag: None,
         }
     }
+
+    pub fn decompose(
+        &self,
+        tile_size: LayerSize,
+        tile_spacing: LayerSize,
+        max_prims: usize,
+    ) -> Vec<LayerPrimitiveInfo> {
+        let mut prims = Vec::new();
```

Comment:
nit: it'd be nice to not have to allocate a vector at all, but we can at least pre-compute its capacity.

Reply:
Vec::new() is guaranteed to not allocate on the heap until something is pushed. So I did it this way since the common case will be no tiling involved. We could pre-calculate a capacity inside the code branch once we determine we *do* need tiling though.